### DISCUSSION TITLE
[Help] How to search for creation date based on journal title

### DISCUSSION THREAD

**Discussion by ray-x** [zk-org/zk#393] [Help]
### Verify

- [X] I searched the existing discussions for help

### How can we help you?

I migrated lots of notes recently and the file creation dates are all shown as yesterday.
If I filter the notes by date, all notes will be shown. So the filter is based on the migration date.
Is it possible to filter notes based on the title? e.g. last week filter shows notes name with the name of `2024-02-25.md` instead of all notes?

**Comment by zalegrala** [comment]
If I use `:ZkNotes` from vim, I can search for a name in the results.  What I don't see is a flag for `zk list` to sort or return only a specific title.

**Comment by ray-x** [comment]
I have a hack for this in note_parse.go
![image](https://github.com/zk-org/zk/assets/1681295/1c78424a-2850-4437-a3c9-644202447191)

I do not think it is optional as the note name can be in other formats as well, e.g. `yyyy-mm-dd` or `dd-mm-yyyy` etc
But currently, zk does not check naming, only the metadata matters. 
This makes it harder for me to migrate large a volume of notes without some shell script to inject metadata.

**Comment by tjex** [comment]
yep, metadata is critical!
  
Upon migration, all notes will be given the current date. However, if you have date metadata in the notes, you can simply delete `.zk/notebook.db` and rerun `ZkIndex`. 

As [YAML dates take precedence over file dates](https://github.com/zk-org/zk/blob/main/docs/note-frontmatter.md), zk will filter the notes correctly by date from then on 👍 

As to a practical solution in your case, you could use the `--match title:'...'` flag.

`zk list --match title:'2024-02-01' to get notes made on the 1st of Feb.

**Comment by ray-x** [comment]
Is there a pattern that can match yyyy-mm-dd? As there are hundreds of notes that need to be migrated.

**Reply by tjex** [reply]
how do you mean? From what I understand you have the date in each notes file name? 
So your 'filter' would just be a title filter instead of a date filter so, `zk list --match title:'2024-03-23' instead of `zk list --created yesterday` (today being the 24th of March)

For future proofing though I would recommend to just write a script that takes the date from the filename and injects it as a `date: <date>` metadata field in all your notes, then reindex.
